test/acceptance/features/editOwnerDetailsValidations.feature

```diff
@@ -0,0 +1,123 @@
+        And I click on the element "button=Search"
+        Then I expect that element "html" contains the text "1 owner record"
+
+    Scenario: Select an owner
+        When I click on the link containing "John Smith"
+        Then I expect that element "h1" contains the text "John Smith"
+        And I expect that element "(//dd)[1]" contains the text "John Smith"
+
+    Scenario: Missing mandatory fields
+        When I click on the link containing "Edit details"
+        When I clear the inputfield "#firstName"
+        And I clear the inputfield "#lastName"
+        And I clear the inputfield "#addressLine1"
+        And I clear the inputfield "#addressLine2"
+        And I clear the inputfield "#town"
+        And I clear the inputfield "#postcode"
+        And I click on the element "button=Save details"
+        Then I expect that element "form" contains the text "Enter a first name" 
+        And I expect that element "form" contains the text "Enter a last name"
+        And I expect that element "form" contains the text "Enter the first line of the address"
+        And I expect that element "form" contains the text "Enter the town or city"
+        And I expect that element "form" contains the text "Enter a postcode"
+
+    Scenario: Invalid owner date of Birth format(day missing)
+        When I set "John" to the inputfield "#firstName"
+        And I set "Smith" to the inputfield "#lastName"
+        And I set "12 INGS LANE" to the inputfield "#addressLine1"
+        And I set "GUISELEY" to the inputfield "#addressLine2"
+        And I set "LEEDS" to the inputfield "#town"
+        And I set "LS20 8DA" to the inputfield "#postcode"
+        And I set "04" to the inputfield "#dateOfBirth-month"
+        And I set "1995" to the inputfield "#dateOfBirth-year"
+        And I click on the element "button=Save details"
+        Then I expect that element "form" contains the text "An owner date of birth must include a day"
+
+    Scenario: Invalid owner date of Birth
+        When I set "10" to the inputfield "#dateOfBirth-day"
+        And I set "13" to the inputfield "#dateOfBirth-month"
+        And I set "1995" to the inputfield "#dateOfBirth-year"
+        And I click on the element "button=Save details"
+        Then I expect that element "form" contains the text "Enter a real date"
+
+    Scenario: Invalid owner date of Birth format(Alphabets)
+        When I set "aa" to the inputfield "#dateOfBirth-day"
+        And I set "bb" to the inputfield "#dateOfBirth-month"
+        And I set "fgbv" to the inputfield "#dateOfBirth-year"
+        And I click on the element "button=Save details"
+        Then I expect that element "form" contains the text "Enter a real date"
+
+    Scenario: Dog Owner must be 16years old
+        When I set "15" to the inputfield "#dateOfBirth-day"
+        And I set "10" to the inputfield "#dateOfBirth-month"
+        And I set "2008" to the inputfield "#dateOfBirth-year"
```

Comment:
It might be worth setting this to say 2010 or later otherwise this test will start failing in 6 months time

Reply:
Updated